pyuvdata/fhd.py

```diff
@@ -283,25 +316,142 @@ def read_fhd(self, filelist, use_model=False, run_check=True, check_extra=True,
+                    # compare with the known_telescopes values
+                    telescope_obj = uvtel.get_telescope(self.telescope_name)
+                    # start warning message
+                    message = ('Telescope location derived from obs lat/lon/alt '
+                               'values does not match the location in the layout file.')
+
+                    if telescope_obj is not False:
+                        if self._latlonalt_close((latitude, longitude, altitude),
+                                                 telescope_obj.telescope_location_lat_lon_alt):
+                            # obs lat/lon/alt matches known_telescopes
+                            message += (' Value from obs lat/lon/alt matches the '
+                                        'known_telescopes values, using them.')
+                            self.telescope_location = location_latlonalt
+                        elif self._xyz_close(arr_center, telescope_obj.telescope_location):
+                            # layout xyz matches known_telescopes
+                            message += (' Value from the layout file matches the '
+                                        'known_telescopes values, using them.')
+                            self.telescope_location = arr_center
+                        else:
+                            # None of the values match each other. Defaulting to known_telescopes value.
+                            message += (' Neither location matches the values '
+                                        'in known_telescopes. Defaulting to '
+                                        'using the known_telescopes values.')
+                            self.telescope_location = telescope_obj.telescope_location
+                    else:
+                        message += (' Telescope is not in known_telescopes. '
+                                    'Defaulting to using the obs derived values.')
+                        self.telescope_location = location_latlonalt
+                    # issue warning
+                    warnings.warn(message)
+            else:
+                self.telescope_location_lat_lon_alt = (latitude, longitude, altitude)
+
+            if 'antenna_coords' in layout_fields:
+                self.antenna_positions = layout['antenna_coords'][0]
+                layout_fields.remove('antenna_coords')
+
+            if 'antenna_names' in layout_fields:
+                self.antenna_names = [uvutils._bytes_to_str(ant).strip() for ant in layout['antenna_names'][0].tolist()]
+                layout_fields.remove('antenna_names')
+            if 'antenna_numbers' in layout_fields:
+                self.antenna_numbers = layout['antenna_numbers'][0]
+                layout_fields.remove('antenna_numbers')
+            if 'n_antenna' in layout_fields:
+                self.Nants_telescope = int(layout['n_antenna'][0])
+                layout_fields.remove('n_antenna')
```

Comment:
sure, I'll add a comment

Reply:
Having some of these missing will cause it to error if the check is run. I can add some warnings to that effect if you think I should.

Reply:
Actually, I just re-read the IDL code. I think most of these (except diameters) are required to be present. So I think I'll take out the conditional altogether.

Reply:
well that simplifies things.

Reply:
yep. I was confused because one part of the code reads them conditionally, but then when it makes the structure, the values have to be present. Hmm... maybe I'll double check with Ian.